src/hooks/useReportIDs.tsx

```diff
@@ -0,0 +1,171 @@
+import {withOnyx} from 'react-native-onyx';
+import type {OnyxCollection, OnyxEntry} from 'react-native-onyx';
+import {getCurrentUserAccountID} from '@libs/actions/Report';
+import * as OptionsListUtils from '@libs/OptionsListUtils';
+import {getPolicyMembersByIdWithoutCurrentUser} from '@libs/PolicyUtils';
+import SidebarUtils from '@libs/SidebarUtils';
+import CONST from '@src/CONST';
+import ONYXKEYS from '@src/ONYXKEYS';
+import type {Beta, Policy, PolicyMembers, PriorityMode, Report, ReportActions, Transaction, TransactionViolation} from '@src/types/onyx';
+import type * as OnyxCommon from '@src/types/onyx/OnyxCommon';
+import useActiveWorkspace from './useActiveWorkspace';
+import useCurrentReportID from './useCurrentReportID';
+import usePermissions from './usePermissions';
+
+type OnyxProps = {
+    chatReports: OnyxCollection<Report>;
+    betas: OnyxEntry<Beta[]>;
+    policies: OnyxCollection<Policy>;
+    allReportActions: OnyxCollection<ReportActions>;
+    transactionViolations: OnyxCollection<TransactionViolation[]>;
+    policyMembers: OnyxCollection<PolicyMembers>;
+    priorityMode: OnyxEntry<PriorityMode>;
+    allTransactions: OnyxCollection<Transaction>;
+};
+
+type WithReportIDsContextProviderProps = OnyxProps & {
+    children: React.ReactNode;
+    currentReportIDForTests?: string;
+};
+
+type ReportIDsContextValue = {
+    orderedReportIDs: string[];
+    reportIDsWithErrors: Record<string, OnyxCommon.Errors>;
+};
+
+const ReportIDsContext = createContext<ReportIDsContextValue>({
+    orderedReportIDs: [],
+    reportIDsWithErrors: {},
+});
+
+function WithReportIDsContextProvider({
+    children,
+    chatReports,
+    betas,
+    policies,
+    allReportActions,
+    transactionViolations,
+    policyMembers,
+    priorityMode,
+    allTransactions,
+    /**
+     * Only required to make unit tests work, since we
+     * explicitly pass the currentReportID in LHNTestUtils
+     * to SidebarLinksData, so this context doesn't have an
+     * access to currentReportID in that case.
+     *
+     * So this is a work around to have currentReportID available
+     * only in testing environment.
+     */
+    currentReportIDForTests,
+}: WithReportIDsContextProviderProps) {
+    const currentReportIDValue = useCurrentReportID();
+    const derivedCurrentReportID = currentReportIDForTests ?? currentReportIDValue?.currentReportID;
+    const {activeWorkspaceID} = useActiveWorkspace();
+    const {canUseViolations} = usePermissions();
+
+    const policyMemberAccountIDs = useMemo(() => getPolicyMembersByIdWithoutCurrentUser(policyMembers, activeWorkspaceID, getCurrentUserAccountID()), [activeWorkspaceID, policyMembers]);
+    const chatReportsKeys = useMemo(() => Object.keys(chatReports ?? {}), [chatReports]);
+    const reportIDsWithErrors = useMemo(
+        () =>
+            // eslint-disable-next-line you-dont-need-lodash-underscore/reduce
+            _.reduce(
```

Comment:
Is there no way to achieve this without the lodash reduce?

Reply:
@mountiny I have actually moved this code from `SidebarLinksData`, so it's not something that's added to the codebase in this PR. If you prefer, I can dedicate some time to achieve the same result without the usage of lodash reduce.

Reply:
Ok in that case feel free to keep it in for now to not add more to the changes made here